module/zfs/dmu.c

```diff
@@ -2337,6 +2337,37 @@ __dmu_object_info_from_dnode(dnode_t *dn, dmu_object_info_t *doi)
 		doi->doi_fill_count += BP_GET_FILL(&dnp->dn_blkptr[i]);
 }
 
+/*
+ * Give the ZFS object, if it contains any dirty nodes
+ * this function flushes all dirty blocks to disk. This
+ * ensures the DMU object info is updated. A more efficient
+ * future version might just find the TXG with the maximum
+ * ID and wait for that to be synced.
+ */
+int
+dmu_object_wait_synced(objset_t *os, uint64_t object)
+{
+	dnode_t *dn;
+	int error, i;
+
+	error = dnode_hold(os, object, FTAG, &dn);
+	if (error) {
```

Comment:
Far be it from me to criticize your ZFS coding style, but it doesn't seem common even in the ZFS code to have single-line "if" blocks with `{}` (and also on line 2364).  Please ignore if incorrect.